stages/org.osbuild.systemd-journald

```diff
@@ -0,0 +1,115 @@
+    },
+    "config": {
+        "additionalProperties": false,
+        "type": "object",
+        "description": "systemd-journald configuration.",
+        "minProperties": 1,
+        "properties": {
+            "Journal": {
+                "additionalProperties": false,
+                "type": "object",
+                "description": "'Journal' configuration section.",
+                "minProperties": 1,
+                "properties": {
+                    "Storage": {
+                        "type": "string",
+                        "description": "Controls where to store journal data.",
+                        "pattern": "^(volatile|persistent|auto|none)$"
+                    },
+                    "Compress": {
+                        "type": "string",
+                        "description": "Sets whether the data objects stored in the journal should be compressed or not. Can also take threshold values.",
+                        "pattern": "^(yes|no|[0-9]+(K|M|G))$"
+                    },
+                    "SplitMode": {
+                        "type": "string",
+                        "description": "Splits journal files per user or to a single file.",
+                        "pattern": "^(uid|none)$"
+                    },
+                    "MaxFileSec": {
+                        "type": "string",
+                        "description": "Max time to store entries in a single file. By default seconds, may be sufixed with units to override this.",
+                        "pattern": "^[0-9]+(year|month|week|day|h|m){0,1}$"
+                    },
+                    "MaxRetentionSec": {
+                        "type": "string",
+                        "description": "Maximum time to store journal entries.",
+                        "pattern": "^[0-9]+(year|month|week|day|h|m){0,1}$"
+                    },
+                    "SyncIntervalSec": {
+                        "type": "integer",
+                        "minimum": 0,
+                        "description": "Timeout before synchronizing journal files to disk."
+                    },
+                    "Audit": {
+                        "type": "string",
+                        "description": "If enabled systemd-journald will turn on kernel auditing on start-up.",
+                        "pattern": "^(yes|no){0,1}$"
+                    }
+                }
+            }
+        }
+    }
+}
+"""
+
+
+def main(tree, options):
+    dropin_filename = options["filename"]
+    dropin_config = options["config"]
+
+    dropin_dir = f"{tree}/usr/lib/systemd/journald.conf.d"
+    os.makedirs(dropin_dir, exist_ok=True)
+
+    config = configparser.ConfigParser()
+    config.optionxform = lambda option: option
+
+    for section, opts in dropin_config.items():
+        if not config.has_section(section):
+            config.add_section(section)
+        for option, value in opts.items():
+            config.set(section, option, str(value))
```

Comment:
Why all this? I think the point of specifying a filename is to drop in a new file. What is the reason to support amending existing files? The point of `*.d/` directories is to never having to amend files, isn't it?

Reply:
Ignore me, I missed that you don't read it in. Sorry. All good!